server/api/apointment_handler.py

```diff
@@ -0,0 +1,62 @@
+from flask import jsonify, Blueprint, request
+import json
+from model.model import Users, Appointments
+from config import db
+from utils.auth.middleware import check_token
+import dateutil
+import re
+
+create_appointment_handler = Blueprint('create_appointment_handler', __name__)
+get_appointment_handler = Blueprint('get_appointment_handler', __name__)
+
+
+@create_appointment_handler.route('/appointment/create', methods=['POST'])
+@check_token
+def create_appointment(user_id):
+    data = json.loads(request.get_data())
+    time = data.get('dateTime')
+    email = data.get('email')
+    timezone = data.get('timezone')
+    # TODO: get related eventType id/url
+
+    if time is None or email is None or timezone is None:
+        return jsonify({'success': False, 'msg': 'Missing field(s)'}), 400
+    if time == "" or email == "" or timezone == "":
+        return jsonify({'success': False, 'msg': 'Empty field(s)'}), 400
+
+    regex = """^(\w|\.|\_|\-)+[@](\w|\_|\-|\.)+[.]\w{2,3}$"""
+    if not re.search(regex, email):
+        return jsonify({'success': False, 'msg': 'Invalid email'}), 400
+
+    try:
+        dateutil.parser.isoparse(time)
+    except:
+        return jsonify({'success': False, 'msg': 'Invalid datetime'}), 400
+
+    user = Users.query.filter_by(id=user_id).first()
+    name = user.name  # need clarify if this name come from FE or the name in DB
+
+    appointment = Appointments(
+        eventType_id="",  # need eventType id
+        name=name,
+        email=email,
+        time=time,
+        timezone=timezone
+    )
+    db.session.add(appointment)
+    try:
+        db.session.commit()
+    except:
+        return jsonify({'success': False, 'msg': 'Failed to record'}), 500
+    return jsonify({'success': True, 'msg': 'Appointment Booked'}), 200
+
```

Comment:
We also need to sync with Google Calendar. For this i think we need to get the duration of event and do something like this: https://developers.google.com/calendar/create-events

make sure to put function under utils/auth/google_client for better organization!

Reply:
Ok, I'll try it later today

Reply:
and also add the sendemail

Reply:
Did you get a chance to add the send email part?

Reply:
No, cuz I realized that the email could be sent from google once the event created. So i was thing if we need sendemail or not. And if we need, i can do it in the next branch, the intergration

Reply:
okk got it